### PR TITLE
Sovereign genesis meta block checker

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- There were some checks in genesis blocks from process component factory for meta blocks to exist
  
## Proposed changes
- Added extra component to check for meta block
- For sovereign shard, meta block should not be check (disabled)

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by raduchis — CHANGES_REQUESTED** [review]

File: cmd/sovereignnode/sovereignNodeRunner.go [hunk 1]
```diff
@@ -1212,6 +1212,7 @@ func (snr *sovereignNodeRunner) CreateManagedProcessComponents(
 		ChainRunType:               common.ChainRunTypeSovereign,
 		ShardCoordinatorFactory:    sharding.NewSovereignShardCoordinatorFactory(),
 		GenesisBlockCreatorFactory: genesisProcess.NewSovereignGenesisBlockCreatorFactory(),
+		GenesisMetaBlockChecker:    processComp.NewDisabledGenesisMetaBlockChecker(),
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply cmd/sovereignnode/sovereignNodeRunner.go:1215]: That check is already being done in `setGenesisHeader`, before calling this func, so we only need it for meta. Will leave it as it is.

Although, thanks to your suggestion, I've decided to add extra check for disabled one(which I've renamed ti sovereign) to not contain meta block

File: factory/processing/interface.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package processing
+
+import "github.com/multiversx/mx-chain-core-go/data"
+
+// GenesisMetaBlockChecker should handle genesis meta block checks after creation
+type GenesisMetaBlockChecker interface {
+	CheckGenesisMetaBlock(genesisBlocks map[uint32]data.HeaderHandler, validatorStatsRootHash []byte) error
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply factory/processing/interface.go:7]: Right, good point, changed it as suggested

File: factory/processing/errors.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package processing
+
+import "errors"
+
+var errGenesisMetaBlockDoesNotExist = errors.New("genesis meta block does not exist")
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply factory/processing/errors.go:5]: Moved errors as suggested

File: cmd/sovereignnode/sovereignNodeRunner.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply cmd/sovereignnode/sovereignNodeRunner.go]: ok

File: factory/processing/errors.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply factory/processing/errors.go]: ok

**Review by raduchis — APPROVED** [review]

**Pull request merged**